website/docs/r/instance.html.markdown

```diff
@@ -77,7 +77,7 @@ instances. See [Shutdown Behavior](https://docs.aws.amazon.com/AWSEC2/latest/Use
 
 * `get_password_data` - (Optional) If true, wait for password data to become available and retrieve it. Useful for getting the administrator password for instances running Microsoft Windows. The password data is exported to the `password_data` attribute. See [GetPasswordData](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_GetPasswordData.html) for more information.
 * `monitoring` - (Optional) If true, the launched EC2 instance will have detailed monitoring enabled. (Available since v0.6.0)
-* `security_groups` - (Optional, EC2-Classic and default VPC only) A list of security group names to associate with.
+* `security_groups` - (Optional, EC2-Classic and default VPC only) A list of security group names or IDs to associate with.
```

Comment:
I think we should specifically call out when to use each 👍  `A list of security group names (EC2-Classic) or IDs (default VPC) to associate with.`

Reply:
Just a quick question, what does `default VPC` mean? Are these the VPC's that are already created?

Reply:
Information about default VPCs can be found at: https://docs.aws.amazon.com/vpc/latest/userguide/default-vpc.html

> If you created your AWS account after 2013-12-04, it supports only EC2-VPC. In this case, you have a default VPC in each AWS Region. A default VPC is ready for you to use so that you don't have to create and configure your own VPC.

In general, yes, they will already exist in newer accounts, but they can also be deleted (usually for security reasons).

Reply:
Ah. It's just that I actually created another VPC so I am confused if that still classifies as a "default VPC".

Reply:
Its only possible to have one default VPC per region. If you have created another VPC its likely that it is not the default VPC for the region. You can verify this in the AWS web console by going to Services > VPC > Your VPCs and looking at the `Default VPC` column.

Reply:
Ah yes. It is not classified as default. I guess what I am leading to is that the VPC I created is not default but using it's ID still worked.

Reply:
If you run `terraform plan` again, does it report any changes that need to happen? If not, that's actually new behavior.

Reply:
Hmm actually yes it does. It says that it needs to recreate my aws instance. 
```
-/+ aws_instance.example (new resource required)
      id:                           "i-0344ac8a8f27efcc2" => <computed> (forces new resource)
```

I actually have a [repo for this](https://github.com/uLan08/play-terraform) if you want to take a look.

Reply:
From the looks of it, what I am doing is wrong and maybe I should have just used the default VPC. I would appreciate it if you could explain why though. Nevertheless, I updated the docs as per your request.

Reply:
> From the looks of it, what I am doing is wrong and maybe I should have just used the default VPC. I would appreciate it if you could explain why though.

The AWS provider is not intended to be opinionated about how you structure your AWS infrastructure, we just follow the API implementation provided by AWS. The code maintainers here will not tell you whether using a default or custom VPC is a better solution for whatever problem you are trying to solve as issues in this GitHub repository are intended only for bugs or feature requests with Terraform's implementation with AWS. For general questions about using AWS, I would reach out to the other [Terraform community forums](https://www.terraform.io/docs/extend/community/index.html) or the [AWS forums](https://forums.aws.amazon.com/).

Reply:
Alright Thanks!